### PR TITLE
Treat PGI compiler flags in nvcc_wrapper

### PR DESCRIPTION
**Pull request by masterleinad**
Alternative to #3239 as suggested in #3252.

### CODE REVIEW

File: bin/nvcc_wrapper [hunk 1]
```diff
@@ -251,6 +251,20 @@ do
     shared_args="$shared_args $std_flag"
     ;;
```
Comment (jjwilke) [inline bin/nvcc_wrapper:253]: I think we need to also revert the changes in kokkos_test_cxx_std.cmake?
````
+    SET(SUPPORTED_NVCC_FLAGS "-std=c++11;-std=c++14;-std=c++17")
+    IF (${KOKKOS_CXX_COMPILER_ID} STREQUAL NVIDIA AND NOT ${${STANDARD_NAME}} IN_LIST SUPPORTED_NVCC_FLAGS)
+      MESSAGE(FATAL_ERROR "CMake wants to use ${${STANDARD_NAME}} which is not supported by NVCC. Using a more recent host compiler or a more recent CMake version might help.")
+    ENDIF()
````

**Review by jjwilke — CHANGES_REQUESTED** [review]

File: bin/nvcc_wrapper (hunk 1 above)
Comment (masterleinad) [reply bin/nvcc_wrapper:253]: Recent `develop` has
```
   IF (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND (KOKKOS_CXX_HOST_COMPILER_ID STREQUAL GNU OR KOKKOS_CXX_HOST_COMPILER_ID STREQUAL Clang))
      SET(SUPPORTED_NVCC_FLAGS "-std=c++11;-std=c++14;-std=c++17")
      IF (NOT ${${STANDARD_NAME}} IN_LIST SUPPORTED_NVCC_FLAGS)
        MESSAGE(FATAL_ERROR "CMake wants to use ${${STANDARD_NAME}} which is not supported by NVCC. Using a more recent host compiler or a more recent CMake version might help.")
      ENDIF()
    ENDIF()
```
which should be fine.

Comment (jjwilke) [reply bin/nvcc_wrapper:253]: Won't this reject PGI + NVCC? Compiler ID will be NVIDIA, but it will have the PGI compiler flag `--c++11`. I could be misunderstanding the code.

Comment (masterleinad) [reply bin/nvcc_wrapper:253]: But the host compiler id is `PGI` and not `GNU` or `Clang`.

Comment (jjwilke) [reply bin/nvcc_wrapper:253]: Got it. I can't read.

**Review by jjwilke — APPROVED** [review]

**Comment by jjwilke** [comment]
I don't have access to a PGI compiler to test this. But looks fine to me.
